api/incidents/views.py

```diff
@@ -54,3 +54,32 @@ def create_redflag_record(self, request):
     def create(self, request):
         request.data['createdBy'] = request.user.id
         return self.validate_record(request)
+
+    def list(self, request):
```

Comment:
combine the methods `list` and `retrieve` into one method

Reply:
For the view set we are using i.e the modelViewSet, we have to separate the two in order to override the default return messages.